snuba/schemas.py

```diff
@@ -44,13 +59,19 @@ def validate_jsonschema(value, schema, set_defaults=True):
     """
     orig = jsonschema.Draft6Validator.VALIDATORS["properties"]
 
-    def validate_and_default(validator, properties, instance, schema):
+    def validate_and_default(
+        validator,
+        properties: Mapping[str, Any],
+        instance: MutableMapping[str, Any],
+        schema,
+    ):
         for property, subschema in properties.items():
-            if "default" in subschema:
+            if property not in instance and "default" in subschema:
                 if callable(subschema["default"]):
-                    instance.setdefault(property, subschema["default"]())
+                    default_value = subschema["default"]()
                 else:
-                    instance.setdefault(property, copy.deepcopy(subschema["default"]))
```

Comment:
This previous implementation was doing a lot of unnecessary copying and calling.